feat: create a specification for the get all uploaded files (#291)
graasp/graasp-account: docs/DisplayUploadedFilesInStorage.md

              "path": "/presentations/Presentation.pdf",
              "parent": null
            }
          ],
          "pagination": {
            "totalItems": 100,
            "totalPages": 5,
            "currentPage": 2,
            "pageSize": 20
          }
        }
    ```
- What happens on failure?
    - Possible Causes of Failure:
        - Authentication Failure: User is not authenticated or token is invalid.
        - Invalid query parameters: The query parameters are missing or invalid.
        - Server Error: Internal server error occurs while processing the request.
  - What error code and error data is used? 
    - 401 Unauthorized:
        ```json 
        {
            "status": "error",
            "error_code": "401",
            "message": "User not authenticated."
        }
        ```
    - 400 Bad request
        ```json 
        {
          "status": "error",
          "message": "Invalid query parameters"
        }
        ```
    - 500 Internal Server Error
        ``` json
        {
            "status": "error",
            "error_code": "500",
            "message": "An unexpected error occurred. Please try again later."
        }

>         ```

Reviewer: Maybe I would reorganise a bit this part like the following. You could even include the successful path. What do you think ?

- Responses:
  - 200 OK
    - The request was successful.
    - schema:
    ```json
    {
        "data": [
          {
            "id":"string", // Item's id
            "name": "string",           // Name of the file
            "size": "number",           // Size of the file in bytes
            "updated": "string",      // Date and time when the item was updated
            "path": "string",           // Location of the file
            "parent": {                 // Parent folder details (if available)
              "id": "string",
              "name": "string"
            }
          }
        ],
        "pagination": {
          "totalItems": 100,           // Total number of items
          "totalPages": 5,             // Total number of pages
          "currentPage": 2,            // Current page number
          "pageSize": 20               // Number of items per page
        }
      }
    ```
    - exemple of response:
    ```...```
   
  - 401 Unauthorized
    - Authentication Failure: User is not authenticated or token is invalid.
    ```json 
        {
            "status": "error",
            "error_code": "401",
            "message": "User not authenticated."
        }
    ```
  - 400 Bad Request
    - Invalid query parameters: The query parameters are missing or invalid.
    ```json 
        {
          "status": "error",
          "message": "Invalid query parameters"
        }
    ```
  - 500 Internal Error
    - Server Error: Internal server error occurs while processing the request.
    ``` json
        {
            "status": "error",
            "error_code": "500",
            "message": "An unexpected error occurred. Please try again later."
        }
    ```

### Frontend design

> - Is this a hook or a mutation? 
>   - This is a hook because we are fetching data from the backend to display in the frontend UI. 

Reviewer: ```suggestion
- A hook is used in the frontend to fetch the data from the backend.
```

> - What are the parameters of the hook?
>   -  Parameters for the hook will include:

Reviewer: ```suggestion
-  Parameters for the hook:
```

      - page: The page number to fetch.
      - pageSize: The number of items per page.

      These parameters will map to the query string parameters used in the backend endpoint (`/members/current/storage/list?page=2&pageSize=20`).

### Backend design

- Where should this go? Is this a new service? Does it extend an already existing service?

>   - This functionality can extend an existing service that manages items. If no such service exists, a new service specifically for managing items (files and folders) should be created.

Reviewer: I don't think this is useful, I'd rather precise in which service the new feature will be implemented if we already know that (for example in the `MemberService`, `ItemService` or any other service).

If we don't know for now, just let like that.
